src/librbd/io/ImageRequestWQ.h

```diff
@@ -107,6 +108,46 @@ class ImageRequestWQ
   std::atomic<unsigned> m_io_blockers { 0 };
   std::atomic<unsigned> m_io_throttled { 0 };
 
+  typedef interval_set<uint64_t> ImageExtents;
+
+  ImageExtents m_in_flight_extents;
+
+  struct BlockedIO {
```

Comment:
Nit: can we not just re-use `ImageDispatcSpec`?

Reply:
Done